### PR TITLE
Fix listener messages duplication

### LINKED ISSUE #106: Fix duplicate listener messages

### PR DESCRIPTION
**Pull request by cheezus1**
resolves #106

### CODE REVIEW

File: lib/core/listener/listener.ex [hunk 1]
```diff
@@ -414,8 +414,17 @@ defmodule Core.Listener do
      }}
   end
 
-  def handle_info(:gc_objects_sent, state) do
-    {:noreply, %{state | objects_sent: [], gc_scheduled: false}}
+  def handle_info(:gc_objects_sent, %{objects_sent: objects_sent} = state) do
+    half_count = floor(Enum.count(objects_sent) / 2)
+
+    gc_objects_sent =
```
Comment (Artur64) [inline lib/core/listener/listener.ex]: Could you explain why exactly we are getting **duplicated** objects here and if  we are sending them, so are we able to fix it at the moment of sending ?

Comment (cheezus1) [reply lib/core/listener/listener.ex]: As several same objects can be received by peers at one time, there were cases where the garbage collection was executed in between these same messages, causing the ones after the GC to be considered new.

Comment (cheezus1) [reply lib/core/listener/listener.ex]: Thus clearing only the earlier objects sent

Comment (thepiwo) [reply lib/core/listener/listener.ex]: I don't understand, so when connecting to multiple peers we receive the same message multiple times?

Comment (thepiwo) [reply lib/core/listener/listener.ex]: so @cheezus1 when we connect to more than 2 nodes, lets say n nodes, wouldn't we eventually receive the message n times? as far as I can see this code only handles the case of exactly 2 messages

Comment (cheezus1) [reply lib/core/listener/listener.ex]: @thepiwo yes, we would, and it would handle n messages as well.

Comment (thepiwo) [reply lib/core/listener/listener.ex]: why specifically the `2` then?

Comment (cheezus1) [reply lib/core/listener/listener.ex]: this stands for the first half of the object sent array

**Pull request merged**